smarts/sstudio/genscenario.py

```diff
@@ -439,25 +439,36 @@ def _validate_entry_tactic(mission):
 
 def gen_traffic_histories(
     scenario: str,
-    histories_datasets: Sequence[str],
+    histories_datasets: Sequence[Union[types.TrafficHistoryDataset, str]],
     overwrite: bool,
 ):
     """Converts traffic history to a format that SMARTS can use.
     Args:
         scenario:
             The scenario directory
         histories_datasets:
-            A sequence of traffic history files.
+            A sequence of traffic history descriptors.
         overwrite:
             If to forcefully write over the previous existing output file
     """
-    genhistories_py = os.path.join(
-        os.path.dirname(os.path.realpath(__file__)), "genhistories.py"
-    )
     for hdsr in histories_datasets:
+        if not isinstance(hdsr, str):
+            from . import genhistories
+
+            genhistories.import_dataset(hdsr, scenario, overwrite)
+            continue
+
```

Comment:
I would prefer to check the specific type rather than accept any other type.
```suggestion
        if isinstance(hdsr, types.TrafficHistoryDataset):
            from . import genhistories

            genhistories.import_dataset(hdsr, scenario, overwrite)
            continue

        assert isinstance(hdsr, str)
```